rootless/rootless.go

```diff
@@ -23,3 +31,45 @@ func RunningWithRootlessKit() bool {
 	})
 	return runningWithRootlessKit
 }
+
+// GetRootlessKitInfo returns the Info object of RootlessKit REST API: https://github.com/rootless-containers/rootlesskit/blob/master/pkg/api/openapi.yaml
+//
+// The object is returned as a json.RawMessage, without unmarshalling into a struct of github.com/rootless-containers/rootlesskit/pkg/api.Info ,
+// so as to retain all the fields that are unknown to github.com/rootless-containers/rootlesskit/pkg/api.Info .
+func GetRootlessKitInfo(ctx context.Context) (json.RawMessage, error) {
+	stateDir := os.Getenv("ROOTLESSKIT_STATE_DIR")
```

Comment:
I realize we may not do this currently for the other binaries, but the `/info` endpoint is already heavy on doing  a _lot_ of things; would it make sense to use a `sync.Once`?

Or are there situations where the rootlesskit binary can be updated _without_ the daemon itself being restarted? (in that case we _could_ use a "dirty" trick to reset the `sync.Once` on SIGHUP (e.g.); I did that for a test in the CLI in;
https://github.com/docker/cli/blob/a32cd16160f1b41c1c4ae7bee4dac929d1484e59/cli/config/config.go#L27

```go
var (
	initConfigDir = new(sync.Once)
```

https://github.com/docker/cli/blob/a32cd16160f1b41c1c4ae7bee4dac929d1484e59/cli/config/config.go#L45-L50

```go
// resetConfigDir is used in testing to reset the "configDir" package variable
// and its sync.Once to force re-lookup between tests.
func resetConfigDir() {
	configDir = ""
	initConfigDir = new(sync.Once)
}
```

Reply:
Info may have dynamic info (in future), so sync.Once doesn't seem correct.